src/lib/auth/README.md

```diff
@@ -0,0 +1,135 @@
+
+- **Auth Router (`src/server/api/routers/auth.ts`):**
+  - This tRPC router provides specific endpoints related to authentication:
+    - `login`: Handles user login by calling `auth.api.signInEmail`.
+    - `logout`: Handles user logout by calling `auth.api.signOut`.
+    - `getSession`: A public procedure to retrieve the current session, primarily for client-side consumption.
+    - `register`: Currently throws a `NOT_IMPLEMENTED` error, as registration is intended to be handled client-side via `authClient.signUp.email`.
+  - A helper `convertToHeaders` is used to adapt Next.js request headers for `better-auth` API calls.
+
+### 4. Frontend State Management
+
+- **Auth Context (`src/contexts/AuthContext.tsx`):**
+  - An `AuthProvider` component wraps the application (or parts of it) to provide authentication state.
+  - It uses `authClient.useSession()` from `better-auth/client` to get the current session information (user, loading state).
+  - A `useAuth` hook is provided for easy access to the `user` object and `isLoading` state in any component.
+
+### 5. UI Components
+
+- **Login & Registration Forms (`src/components/auth/LoginForm.tsx`, `src/components/auth/RegisterForm.tsx`):**
+
+  - These React components provide the user interface for signing in and signing up.
+  - They use methods from the `authClient` (`authClient.signIn.email` and `authClient.signUp.email`) to interact with the `better-auth` backend.
+  - After successful operations, they typically invalidate the `auth.getSession` tRPC query to refresh the `AuthContext`.
+
+- **Auth Tile (`src/app/map/[id]/Tile/auth.dynamic.tsx`):**
+
+  - A dynamically imported component that hosts the `LoginForm` and `RegisterForm`.
+  - It allows users to switch between login and registration views.
+  - This tile is likely intended to be displayed prominently when a user is not authenticated.
+
+- **User Navigation (`src/components/layout/UserNav.tsx` - as per plan):**
+  - This component would display user information and a logout button.
+  - Logout is handled using `authClient.signOut()`.
+
+### 6. Database Integration
+
+- **Drizzle Schema (`src/server/db/schema/`):**
+  - `better-auth` requires specific tables for its operation: `users`, `accounts`, `sessions`, and `verificationTokens` (or `verification`).
+  - The `drizzleAdapter` in `src/server/auth.ts` maps `better-auth`'s expected schema names (singular) to our actual Drizzle table names (plural, e.g., `schema.users`, `schema.accounts`).
+  - Existing tables like `hexMaps` have their `ownerId` field updated to reference the `users.id` table to associate data with authenticated users.
+
+## Authentication Flow
+
+1.  **Registration:**
+
+    - The user fills out the `RegisterForm`.
+    - `authClient.signUp.email` is called on the client-side.
+    - `better-auth` handles user creation, password hashing, and potentially sends a verification email (depending on configuration).
+    - The client's session is updated.
```

Comment:
After registration, the user's map is also created and the user is redirected to her map